### PR TITLE
Empty Stats, Publicize: add the option to enable the publicize module if not active

### LINKED ISSUE #17309: Empty Stats: Investigate publicize with Jetpack sites
When tapping "Sharing" (both from the Empty Stats nudge and My Site) the Publicize screen is shown for .com sites and for Jetpack sites the "Manage sharing buttons" are displayed.

### PR DESCRIPTION
**Pull request by leandroalonso**
Fixes #17309

### To test

1. Login with an account that has a Jetpack site without the `publicize` module enabled
2. For this Jetpack site, go to **Stats**
3. When the Publicize nudge appears tap on **Enable post sharing**
4. The screen should say that Publicize is not active, tap **Enable**
5. The regular Sharing screen should appear
6. Connect any social network to make sure it's working
7. Go back to **Stats**
8. Open the screen again, no message about the module should appear

Additional info:

* You can turn off the wi-fi to test the error
* You can activate/deactivate the Publicize module white testing. Just go to `yoursite/wp-admin/admin.php?page=jetpack_modules`

### Screenshots

| With the module deactivated | With the module active |
| --- | --- |
| Simulator Screen Shot - iPhone 13 - 2021-10-14 at 11 46 19 | Simulator Screen Shot - iPhone 13 - 2021-10-14 at 12 27 05 |

### Additional notes

I've added a new entity that works like a paywall but for Jetpack modules. We can even refactor Stats to use this one — I'll do that in another PR once that lands in `develop`.

The idea is that any VC can use it to block a feature until the module is enabled.

## Regression Notes
1. Potential unintended areas of impact
-

2. What I did to test those areas of impact (or what existing automated tests I relied on)
-

3. What automated tests I added (or what prevented me from doing so)
I'll add unit tests for the `JetpackModuleHelper` but want to make this PR available for review ASAP.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blog/JetpackModuleHelper.swift [hunk 1]
```diff
@@ -0,0 +1,93 @@
+import Foundation
+import UIKit
+
+public typealias JetpackModuleHelperViewController = JetpackModuleHelperDelegate & UIViewController
+
+@objc public protocol JetpackModuleHelperDelegate: AnyObject {
+    func jetpackModuleEnabled()
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/JetpackModuleHelper.swift:7]: What do you think of renaming this to `didEnableJeptackModule`? I think it makes it slightly more clear.

**Review by momo-ozawa — COMMENTED** [review]
Works as described.

I wasn't able to test the publicize connection flow, as that's not working for me right now. :(
> Connect any social network to make sure it's working

**Comment by momo-ozawa** [comment]
Also, I'm getting a diff for `Podfile.lock` when I run `rake dependencies` 🤔

File: WordPress/Classes/ViewRelated/Blog/JetpackModuleHelper.swift (hunk 1 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Blog/JetpackModuleHelper.swift:7]: Done in bd5682f363c63ce3b048819554fbe70e4d6def64

**Comment by nikola-milicevic** [comment]
>Works as described.
I wasn't able to test the publicize connection flow, as that's not working for me right now. :(

I can confirm this. Same situation. Although I could test the no internet error 😊
LGTM.

**Review by nikola-milicevic — APPROVED** [review]

**Pull request merged**